### PR TITLE
kubeadm: Make kubeadm use the right CSR approver for the right version

### LINKED ISSUE #289: Deploying v1.6 clusters with kubeadm at HEAD is broken
https://github.com/kubernetes/kubernetes/pull/45619 completely removed the flag we used for auto-approval in v1.6 (`--insecure-experimental-approve-all-kubelet-csrs-for-group=system:bootstrappers`)

This "broke" deploying v1.6 clusters with kubeadm in the sense that `kubeadm join` will wait until you manually approve the CSR as there is no autoapprover configured.

The fix is pretty simple, we just need to check the version and use the right arguments for the right versions respectively.

PR: https://github.com/kubernetes/kubernetes/pull/46864

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

fixes regression caused in: https://github.com/kubernetes/kubernetes/pull/45619

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes: https://github.com/kubernetes/kubeadm/issues/289

**Special notes for your reviewer**:

cc @pipejakob our e2e CI should probably go green after this change

**Release note**:

```release-note
NONE
```
@mikedanese @pipejakob @timothysc @liggitt

### CODE REVIEW

**Comment by pipejakob** [comment]
@luxas Cool, but as a heads-up, our CI job was failing because of quota issues (fixed now), but hasn't rerun in a while because the submit queue is a little broken now (it seems that someone manually merged some commits that broke verification). No new commits have been merged into kubernetes for 16 hours or so, so it hasn't run in that long. That's not to say there weren't also other regressions caused by #45619 that might appear now, though.

I believe @spxtr is the test-infra on-call and is looking into unblocking the submit queue.

**Comment by pipejakob** [comment]
Actually, looks like https://github.com/kubernetes/kubernetes/pull/46863 should take care of it.

**Comment by liggitt** [comment]
it wasn't a merge that broke master, it was a new git tag :-/

File: cmd/kubeadm/app/master/manifests.go [hunk 1]
```diff
@@ -428,6 +429,11 @@ func getControllerManagerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted
 		"use-service-account-credentials":  "true",
 		"controllers":                      "*,bootstrapsigner,tokencleaner",
 	}
+	if k8sVersion.LessThan(kubeadmconstants.MinimumCSRSARApproverVersion) {
+		// enable the former CSR group approver for v1.6 clusters.
+		// TODO(luxas): Remove this once we're targeting v1.8 at HEAD
+		defaultArguments["insecure-experimental-approve-all-kubelet-csrs-for-group"] = bootstrapapi.BootstrapGroup
```
Comment (liggitt) [inline cmd/kubeadm/app/master/manifests.go]: how was this being set before?

**Comment by pipejakob** [comment]
@liggitt Yeah, Joe pointed out https://github.com/kubernetes/kubernetes/pull/46863 to me. Thanks for the fix!

File: cmd/kubeadm/app/master/manifests.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/master/manifests.go]: @liggitt It was being set unconditionally in the `defaultArguments` slice initializer above. Now only for v1.6

Comment (liggitt) [inline cmd/kubeadm/app/master/manifests.go]: ah, and got removed by the SAR PR. ok.

**Comment by liggitt** [comment]
this LGTM, but bazel is very unhappy

**Comment by luxas** [comment]
I'll fixup the tests tomorrow or so, late here now.
@mikedanese it would be awesome if I could get a soft approval on this and be able to self-apply once I have generated code and tests uploaded.

**Comment by luxas** [comment]
@pipejakob Is there a way to invoke the kubeadm CI with a (@)k8s-bot command btw?

**Comment by pipejakob** [comment]
@k8s-bot pull-kubernetes-e2e-kubeadm-gce test this

**Comment by pipejakob** [comment]
Oh, the kubeadm pull job actually depends on a successful completion of the bazel pull job (that's the step that builds/uploads the .debs we use), so I can't successfully trigger it until you have `pull-kubernetes-bazel` passing.

File: cmd/kubeadm/app/master/manifests.go [hunk 2]
```diff
@@ -54,7 +55,7 @@ const (
 )
 
 var (
-	v170 = version.MustParseSemantic("v1.7.0-alpha.0")
+	v170alpha0 = version.MustParseSemantic("v1.7.0-alpha.0")
```
Comment (timothysc) [inline cmd/kubeadm/app/master/manifests.go]: I actually prefer the v170 b/c we should only support released versions, correct?

**Comment by liggitt** [comment]
@luxas are you able to update this to fix bazel and tests?

**Comment by luxas** [comment]
@liggitt Will do in a couple of hours. Sorry for the slight delay, am partially/have been OOO.

**Comment by luxas** [comment]
@k8s-bot pull-kubernetes-e2e-kubeadm-gce test this

**Comment by liggitt** [comment]
would like to see a green CI run - https://k8s-gubernator.appspot.com/pr/46864

**Comment by luxas** [comment]
/approve no-issue

I must to this due to https://github.com/kubernetes/test-infra/issues/2975

**Comment by liggitt** [comment]
the item linked in the description isn't an issue, it's another PR

**Pull request merged**